### DISCUSSION TITLE
Pimcore 11 web2print gotenberg infinite loading

### DISCUSSION THREAD

**Discussion by DGrosmayer** [pimcore/pimcore#15364] [Q&A / Support] [answered]
Hello Pimcore Community,

im currently trying out the Web2Print functionality on Pimcore 11 and have installed the skeleton for pimcore 11  for that. I started gotenberg with docker and also got correct responses when I try to send there any html through Postman - it also worked when I try the "test" button on the web2print Settings in Pimcore backend. However if I try to click on Generate PDF on the example Printpage, it stays on 0% all the time and on the network tab there is always a new "active-generate-process" generated. Is there a way to fix this?

![image](https://github.com/pimcore/pimcore/assets/110802714/b8570cf2-9263-4027-b136-46564a4fb93f)

![image](https://github.com/pimcore/pimcore/assets/110802714/426457ae-a16f-4a4d-b868-40badc83f7a8)

![image](https://github.com/pimcore/pimcore/assets/110802714/201476ec-5d18-445e-9375-879cf7a821e5)

![image](https://github.com/pimcore/pimcore/assets/110802714/a149141a-7b6a-49d4-af4e-d6df788d6065)

Best Regards
Daniel

**Comment by kingjia90** [comment, marked as answer]
Hi Daniel,

looks like that the maintenance cronjob is not set
https://pimcore.com/docs/platform/Pimcore/Getting_Started/Installation/Webserver_Installation#5-maintenance-cron-job

Let us know if it works after setting it

Best regards,
JJ

**Reply by DGrosmayer** [reply]
Hi JJ,

thank you for your reply. With your suggestion I managed to fix the "0% Issue" I had. Unfortunately now I have errors regarding the generation itself.

After a while Gotenberg returns a statuscode 200 but the maintenance job returns multiple "empty replies from server". 

```
16:40:37 ERROR     [pimcore] Exception: Error during PDF-Generation:cURL error 52: Empty reply from server (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://gotenberg:3000/forms/chromium/convert/html in /var/www/my_pimcore_project/my-project/vendor/pimcore/web-to-print-bundle/src/Processor/Gotenberg.php:69
Stack trace:
#0 /var/www/my_pimcore_project/my-project/vendor/pimcore/web-to-print-bundle/src/Processor.php(117): Pimcore\Bundle\WebToPrintBundle\Processor\Gotenberg->buildPdf()
```

If I check on the PDF that was generated I get non-existant images.

![image](https://github.com/pimcore/pimcore/assets/110802714/f527f815-41bb-4232-9d5c-0ac6849cd7a4)
![image](https://github.com/pimcore/pimcore/assets/110802714/52885d9a-e8a5-4661-bc94-b1628c6d158b)

 If I hover over the links I get something like this:
`http://gotenberg:3000/en/shop/Products/Cars/Sports-Cars/Jaguar-E-Type~p12`

Inside the debug messages of gotenberg it says that he cant load the resource. 
So gotenberg tries to get the images from the relative url of himself instead of Pimcore. 
Is there a setting I have to set in Pimcore for that?

`{"level":"debug","ts":1686926845.368053,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"Log.entryAdded\",\"params\":{\"entry\":{\"source\":\"network\",\"level\":\"error\",\"text\":\"Failed to load resource: net::ERR_NAME_NOT_RESOLVED\",\"timestamp\":1.686926845366973e+12,\"url\":\"http://gotenberg:3000/Car%20Images/jaguar/6/image-thumb__6__print_productcell/Jaguar_E-Type_Series_1_3.8_Litre_1961.ba5ab45a.jpg\",\"networkRequestId\":\"48.12\"}},\"sessionId\":\"BD4709F8B6463C01A7DE9E4BA27A39B2\"}","trace":"4a9954ad-42f2-4c93-9963-be4e329bb8a4"}`


Best Regards
Daniel

**Reply by kingjia90** [reply]
Looks like we may need this PR to be merged first
https://github.com/pimcore/web-to-print-bundle/pull/33
Do you have chance to try it out and see if it works?

**Comment by betterapp** [comment]
I have exactly the same problem as @DGrosmayer 

I setup Gotenberg via Docker- test works fine.
maintenance cronjob is also set

```
*/5 * * * * /usr/bin/php8.2 /home/daniel/Web/pimcore/bin/console pimcore:maintenance
*/5 * * * * /usr/bin/php8.2 /home/daniel/Web/pimcore/bin/console messenger:consume pimcore_core pimcore_maintenance pimcore_scheduled_tasks pimcore_image_optimize pimcore_search_backend_message --time-limit=300
*/5 * * * * /usr/bin/php8.2 /home/daniel/Web/pimcore/bin/console messenger:consume datahub_es_index_queue --limit=5000 --time-limit=240
*/5 * * * * /usr/bin/php8.2 /home/daniel/Web/pimcore/bin/console messenger:consume pimcore_asset_update --limit=5000 --time-limit=240

```

The generate process stays on 0% all the time and on the network tab there is always a new "active-generate-process" generated

When I click on "Generate PDF" it make request to:
![image](https://github.com/pimcore/pimcore/assets/7449124/dc80e805-dd05-42d2-b079-e24d4312c36b)

but it does not return json response:
![image](https://github.com/pimcore/pimcore/assets/7449124/21f9fb68-944f-4132-9311-2fb7174f16b9)

On demo.pimcore.com I see that it return 
![image](https://github.com/pimcore/pimcore/assets/7449124/237b1b87-803c-41c0-966a-0bc2ec7189c1)
